File: Ruby/Pages/GoogleHomePage.rb
#! ruby -Ku

require 'selenium-webdriver'
require File.dirname(__FILE__) + '/PageObject'
require File.dirname(__FILE__) + '/GoogleResultPage'

class GoogleHomePage < PageObject
  def initialize(browser) 
    super(browser)
    @key_text = find_element_by_name("q")
  end

  def input_key(value)
    input_value(@key_text, value)
    return self
  end

  def Search()
    Submit(@key_text)
    return GoogleResultPage.new(@driver)
  end
end

File: Python/Pages/GoogleHomePage.py
# -*- coding: utf-8 -*-
from Pages.GoogleResultPage         import GoogleResultPage
from Pages.PageObject               import PageObject

class GoogleHomePage(PageObject):
    def __init__(self, driver):
        PageObject.__init__(self, driver)
        self.get("http://www.google.com")
        self.key_text = self.find_element_by_name("q")

    def input_key(self, value):
        self.input_value(self.key_text, value)
        return self

    def Search(self):
        self.Submit(self.key_text)
        return GoogleResultPage(self.driver)
